vec-lib/main.rkt

```diff
@@ -69,26 +69,25 @@
          lerp)
 
 
-(define (vec size . params)
+(define (vec size param0 params)
   (if (and
-       (number? (car params))
-       (null? (cdr params)))
-      (for/list ([i (in-range size)])
-        (car params))
-      (let ([params (append (flatten params) (list 0. 0. 0. 0.))])
+       (number? param0)
+       (null? params))
+      (make-list size param0)
+      (let ([params (flatten (list* param0 params '(0. 0. 0. 0.)))])
         (take params size))))
```

Comment:
Since you're using `flatten` anyway, `append` would just do duplicate work. Using `'(0. 0. 0. 0.)` rather than `list` avoids allocating a new list each time.